docs/blog/_posts/2021-02-17-scala3-rc1.md

```diff
@@ -0,0 +1,242 @@
+
+```scala
+import scala.annotation.*  // imports everything in the annotation package
+```
+
+Renaming operator `=>` is replaced by a soft keyword `as`. `as` is also allowed outside braces. For example:
+
+```scala
+import scala.collection.mutable as mut
+import NumPy as np
+```
+
+For the details and discussion, see [PR #11244](https://github.com/lampepfl/dotty/pull/11244). Read more about this change in the [documentation](https://dotty.epfl.ch/docs/reference/changed-features/imports.html).
+
+# Use `uninitialized` for wildcard initializers
+An obscure use of `_` occurs in var definitions:
+
+```scala
+var x: T = _
+```
+
+It defines a concrete variable x without an initial value, or rather the default initial value that the JVM assigns to object fields. It can only be used in a class or object, not to initialize a local variable.
+
+We came up with an arguably better way to express this idiom: the special `uninitialized` value in the `scala.compiletime` object. To get an uninitialized field, you now write:
+
+```scala
+import scala.compiletime.uninitialized
+
+var x: A = uninitialized
+```
+
+This way expresses the intent of the idiom in a more verbose and easy to read way than simply writing an underscore.
+
+For discussion, see [PR #11231](https://github.com/lampepfl/dotty/pull/11231), and the [documentation](https://dotty.epfl.ch/docs/reference/dropped-features/wildcard-init.html) is available on our website.
+
+# Allow `with` after class
+[PR #11157](https://github.com/lampepfl/dotty/pull/11157) brings an syntactic change whereby you have to write `with` instead of `:` to signify the start of the body of a class definition. For example:
+
+```scala
+object Foo with  // Was previously `object Foo:`
+  val x = 10
+```
+
+# Eta-expand companion object if functions are expected
+Starting from RC1, we no longer generate a function parent for companions of case classes. Which means, for example, that given `case class Foo(x: Int)`, you won't be able to use `Foo` in a position where a function is expected:
+
+```scala
+case class Foo(x: Int)
+def f(g: Int => Foo) = g(10)
+
+f(Foo)
+```
+
+Results in:
+
+```scala
+1 |f(Foo)
+  |  ^^^
+  |The method `apply` is inserted. The auto insertion will be deprecated, please write `Foo.apply` explicitly.
+```
+
+As the warning suggests, now you should write `Foo.apply` instead of `Foo`. See [Issue #6190](https://github.com/lampepfl/dotty/issues/6190) and [PR #7207](https://github.com/lampepfl/dotty/pull/7207) for discussion.
+
+# Settling on the `scaladoc` as the documentation tool
+We have settled on using `scaladoc` as the documentation build tool. The obsolete `dotty-doc` (or `scala3-doc`) is removed in RC1. We have also removed all the Kotlin dependencies (Dokka, etc.) from scaladoc. For details, see [PR #11349](https://github.com/lampepfl/dotty/pull/11349). To read more about `scaladoc`, see [documentation](https://dotty.epfl.ch/docs/usage/scaladoc/index.html)
+
+
```

Comment:
Add:

# Use `future` and `future-migration` to specify language versions after 3.0 in `-source`

[PR #11355] changes the `-source` specifier for the Scala version(s) after 3.0 from `3.1` to future. I.e. it is now
`-source future` and `-source future-migration` instead of `-source 3.1` and `-source 3.1-migration`. Language imports are changed analogously. The reason for the change is that we want to keep the possibility open to ship a `3.1` version that does not yet contain all the changes enabled under `-source future`.